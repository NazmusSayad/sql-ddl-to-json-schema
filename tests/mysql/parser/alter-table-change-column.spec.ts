import { getParsedFormat } from '../parse-handler';
import { run } from '../runner';

run(getParsedFormat, {
  'Parser: Should alter table changing column name and definition with options and position after another column.': {
    queries: [
      `ALTER TABLE people change column name fullname varchar(255) character set utf8 default 'John Smith' after ssn;`,
      `ALTER TABLE people change name fullname varchar(255) character set utf8 default 'John Smith' after ssn;`,
    ],
  },

  'Parser: Should alter table changing column name and definition with option, first position.': {
    queries: [
      `ALTER TABLE people change column name fullname varchar(255) character set utf8 first;`,
      `ALTER TABLE people change name fullname varchar(255) character set utf8 first;`,
    ],
  },

  'Parser: Should alter table changing column name and definition.': {
    queries: [
      `ALTER TABLE people change column name fullname varchar(255);`,
      `ALTER TABLE people change name fullname varchar(255);`,
    ],
  },

  'Parser: Should alter table modifying column definition with options and position after another column.': {
    queries: [
      `ALTER TABLE people modify column name varchar(255) character set utf8 default 'John Smith' after ssn;`,
      `ALTER TABLE people modify name varchar(255) character set utf8 default 'John Smith' after ssn;`,
    ],
  },

  'Parser: Should alter table changing column definition with option, first position.': {
    queries: [
      `ALTER TABLE people modify column name varchar(255) character set utf8 first;`,
      `ALTER TABLE people modify name varchar(255) character set utf8 first;`,
    ],
  },

  'Parser: Should alter table changing column definition.': {
    queries: [
      `ALTER TABLE people modify column name varchar(255);`,
      `ALTER TABLE people modify name varchar(255);`,
    ],
  },

  'Parser: Should alter table dropping column.': {
    queries: [
      `ALTER TABLE people drop column if exists middle_name;`,
      `ALTER TABLE people drop if exists middle_name;`,
      `ALTER TABLE people drop column middle_name;`,
      `ALTER TABLE people drop middle_name;`,
    ],
  },
});
