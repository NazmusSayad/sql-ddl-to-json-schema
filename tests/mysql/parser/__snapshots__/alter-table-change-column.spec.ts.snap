// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser: Should alter table changing column definition with option, first position. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeColumn",
                    "column": "name",
                    "datatype": {
                      "id": "O_DATATYPE",
                      "def": {
                        "id": "O_VARIABLE_STRING_DATATYPE",
                        "def": {
                          "datatype": "varchar",
                          "length": 255
                        }
                      }
                    },
                    "columnDefinition": [
                      {
                        "id": "O_COLUMN_DEFINITION",
                        "def": {
                          "charset": "utf8"
                        }
                      }
                    ],
                    "position": {
                      "after": null
                    }
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table changing column definition with option, first position. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeColumn",
                    "column": "name",
                    "datatype": {
                      "id": "O_DATATYPE",
                      "def": {
                        "id": "O_VARIABLE_STRING_DATATYPE",
                        "def": {
                          "datatype": "varchar",
                          "length": 255
                        }
                      }
                    },
                    "columnDefinition": [
                      {
                        "id": "O_COLUMN_DEFINITION",
                        "def": {
                          "charset": "utf8"
                        }
                      }
                    ],
                    "position": {
                      "after": null
                    }
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table changing column definition. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeColumn",
                    "column": "name",
                    "datatype": {
                      "id": "O_DATATYPE",
                      "def": {
                        "id": "O_VARIABLE_STRING_DATATYPE",
                        "def": {
                          "datatype": "varchar",
                          "length": 255
                        }
                      }
                    },
                    "columnDefinition": [],
                    "position": null
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table changing column definition. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeColumn",
                    "column": "name",
                    "datatype": {
                      "id": "O_DATATYPE",
                      "def": {
                        "id": "O_VARIABLE_STRING_DATATYPE",
                        "def": {
                          "datatype": "varchar",
                          "length": 255
                        }
                      }
                    },
                    "columnDefinition": [],
                    "position": null
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table changing column name and definition with option, first position. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeColumn",
                    "column": "name",
                    "newName": "fullname",
                    "datatype": {
                      "id": "O_DATATYPE",
                      "def": {
                        "id": "O_VARIABLE_STRING_DATATYPE",
                        "def": {
                          "datatype": "varchar",
                          "length": 255
                        }
                      }
                    },
                    "columnDefinition": [
                      {
                        "id": "O_COLUMN_DEFINITION",
                        "def": {
                          "charset": "utf8"
                        }
                      }
                    ],
                    "position": {
                      "after": null
                    }
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table changing column name and definition with option, first position. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeColumn",
                    "column": "name",
                    "newName": "fullname",
                    "datatype": {
                      "id": "O_DATATYPE",
                      "def": {
                        "id": "O_VARIABLE_STRING_DATATYPE",
                        "def": {
                          "datatype": "varchar",
                          "length": 255
                        }
                      }
                    },
                    "columnDefinition": [
                      {
                        "id": "O_COLUMN_DEFINITION",
                        "def": {
                          "charset": "utf8"
                        }
                      }
                    ],
                    "position": {
                      "after": null
                    }
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table changing column name and definition with options and position after another column. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeColumn",
                    "column": "name",
                    "newName": "fullname",
                    "datatype": {
                      "id": "O_DATATYPE",
                      "def": {
                        "id": "O_VARIABLE_STRING_DATATYPE",
                        "def": {
                          "datatype": "varchar",
                          "length": 255
                        }
                      }
                    },
                    "columnDefinition": [
                      {
                        "id": "O_COLUMN_DEFINITION",
                        "def": {
                          "charset": "utf8"
                        }
                      },
                      {
                        "id": "O_COLUMN_DEFINITION",
                        "def": {
                          "default": "John Smith"
                        }
                      }
                    ],
                    "position": {
                      "after": "ssn"
                    }
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table changing column name and definition with options and position after another column. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeColumn",
                    "column": "name",
                    "newName": "fullname",
                    "datatype": {
                      "id": "O_DATATYPE",
                      "def": {
                        "id": "O_VARIABLE_STRING_DATATYPE",
                        "def": {
                          "datatype": "varchar",
                          "length": 255
                        }
                      }
                    },
                    "columnDefinition": [
                      {
                        "id": "O_COLUMN_DEFINITION",
                        "def": {
                          "charset": "utf8"
                        }
                      },
                      {
                        "id": "O_COLUMN_DEFINITION",
                        "def": {
                          "default": "John Smith"
                        }
                      }
                    ],
                    "position": {
                      "after": "ssn"
                    }
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table changing column name and definition. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeColumn",
                    "column": "name",
                    "newName": "fullname",
                    "datatype": {
                      "id": "O_DATATYPE",
                      "def": {
                        "id": "O_VARIABLE_STRING_DATATYPE",
                        "def": {
                          "datatype": "varchar",
                          "length": 255
                        }
                      }
                    },
                    "columnDefinition": [],
                    "position": null
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table changing column name and definition. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeColumn",
                    "column": "name",
                    "newName": "fullname",
                    "datatype": {
                      "id": "O_DATATYPE",
                      "def": {
                        "id": "O_VARIABLE_STRING_DATATYPE",
                        "def": {
                          "datatype": "varchar",
                          "length": 255
                        }
                      }
                    },
                    "columnDefinition": [],
                    "position": null
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table dropping column. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "dropColumn",
                    "column": "middle_name"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table dropping column. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "dropColumn",
                    "column": "middle_name"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table dropping column. 3`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "dropColumn",
                    "column": "middle_name"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table dropping column. 4`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "dropColumn",
                    "column": "middle_name"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table modifying column definition with options and position after another column. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeColumn",
                    "column": "name",
                    "datatype": {
                      "id": "O_DATATYPE",
                      "def": {
                        "id": "O_VARIABLE_STRING_DATATYPE",
                        "def": {
                          "datatype": "varchar",
                          "length": 255
                        }
                      }
                    },
                    "columnDefinition": [
                      {
                        "id": "O_COLUMN_DEFINITION",
                        "def": {
                          "charset": "utf8"
                        }
                      },
                      {
                        "id": "O_COLUMN_DEFINITION",
                        "def": {
                          "default": "John Smith"
                        }
                      }
                    ],
                    "position": {
                      "after": "ssn"
                    }
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table modifying column definition with options and position after another column. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeColumn",
                    "column": "name",
                    "datatype": {
                      "id": "O_DATATYPE",
                      "def": {
                        "id": "O_VARIABLE_STRING_DATATYPE",
                        "def": {
                          "datatype": "varchar",
                          "length": 255
                        }
                      }
                    },
                    "columnDefinition": [
                      {
                        "id": "O_COLUMN_DEFINITION",
                        "def": {
                          "charset": "utf8"
                        }
                      },
                      {
                        "id": "O_COLUMN_DEFINITION",
                        "def": {
                          "default": "John Smith"
                        }
                      }
                    ],
                    "position": {
                      "after": "ssn"
                    }
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;
